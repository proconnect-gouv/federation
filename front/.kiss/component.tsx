import React from 'react';

interface DumbComponentProps {
  className: string;
}

export const DumbComponent: React.FC<DumbComponentProps> = React.memo(
  ({ className }: DumbComponentProps) => {
    return <div />;
  },
);

DumbComponent.displayName = 'DumbComponent';
