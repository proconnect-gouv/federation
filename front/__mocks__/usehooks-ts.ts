export const useMediaQuery = jest.fn();

export const useDocumentTitle = jest.fn();

export const useLocalStorage = jest.fn(() => [expect.any(String), jest.fn(), jest.fn()]);

export const useToggle = jest.fn(() => [
  expect.any(Boolean),
  jest.fn().mockImplementation((v) => !v),
  jest.fn(),
]);

export const useCopyToClipboard = jest.fn(() => [expect.any(String), jest.fn()]);
