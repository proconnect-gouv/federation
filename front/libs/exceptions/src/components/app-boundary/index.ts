/* istanbul ignore file */

// declarative file
export * from './app-boundary.component';
