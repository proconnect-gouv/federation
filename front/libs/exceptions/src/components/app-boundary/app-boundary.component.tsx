import type { FallbackProps } from 'react-error-boundary';
import { useDocumentTitle } from 'usehooks-ts';

import { LogoFranceConnect } from '@fc/assets';

// @TODO
// - translate with i18n
// - find a way to not duplicate code from @fc/dsfr
export const AppBoundaryComponent = ({ error }: FallbackProps) => {
  useDocumentTitle('FranceConnect - Erreur');

  // @NOTE TSError : img.src should be a string
  const logo = LogoFranceConnect as unknown as string;

  return (
    <div className="sticky-body">
      <div className="sticky-content">
        {/* <!-- APPLICATION HEADER --> */}
        <header className="fr-header" role="banner">
          <div className="fr-header__body">
            <div className="fr-container">
              <div className="fr-header__body-row">
                <div className="fr-header__brand fr-enlarge-link">
                  <div className="fr-header__brand-top">
                    <div className="fr-header__logo">
                      <a
                        aria-disabled="true"
                        href="/"
                        style={{ cursor: 'pointer' }}
                        title="R√©publique Fran√ßaise">
                        <p className="fr-logo">
                          R√©publique
                          <br />
                          Fran√ßaise
                        </p>
                      </a>
                    </div>
                    <div className="fr-header__operator">
                      <img alt="FranceConnect" className="fr-responsive-img" src={logo} />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </header>
        {/* <!-- endof APPLICATION HEADER --> */}
        <main className="fr-container fr-my-8w">
          {/* <!-- APPLICATION CONTENT --> */}
          <h1>Oooops, Something went wrong ! ü§∑</h1>
          <div className="fr-alert fr-alert--error fr-alert--md">
            <h3 className="fr-alert__title">{error.message}</h3>
            {error && (
              <ul>
                {error.stack.split('\n').map((v: string, index: number) => (
                  // eslint-disable-next-line react/no-array-index-key
                  <li key={`error-stack::${index}`}>{v}</li>
                ))}
              </ul>
            )}
          </div>
          {/* <!-- endof APPLICATION CONTENT --> */}
        </main>
      </div>
      {/* <!-- APPLICATION FOOTER --> */}
      <footer className="sticky-footer fr-footer" id="footer" role="contentinfo">
        <div className="fr-container">
          <div className="fr-footer__body">
            <div className="fr-footer__brand fr-enlarge-link">
              <p className="fr-logo">
                R√©publique
                <br />
                Fran√ßaise
              </p>
              <a
                aria-disabled="true"
                className="fr-footer__brand-link"
                href="/"
                style={{ cursor: 'pointer' }}
                title="R√©publique Fran√ßaise">
                <img
                  alt="FranceConnect"
                  className="fr-footer__logo fr-responsive-img"
                  src={logo}
                  style={{ height: 90, maxHeight: 90 }}
                />
              </a>
            </div>
          </div>
        </div>
      </footer>
      {/* <!-- endof APPLICATION FOOTER --> */}
    </div>
  );
};
