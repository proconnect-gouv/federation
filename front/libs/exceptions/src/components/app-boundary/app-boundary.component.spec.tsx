import { render } from '@testing-library/react';
import { useDocumentTitle } from 'usehooks-ts';

import { AppBoundaryComponent } from './app-boundary.component';

describe('AppBoundaryComponent', () => {
  beforeEach(() => {
    // @NOTE
    // prevent console.error to be displayed in the console
    // due to SVG Logo as string into img.src attribute
    jest.spyOn(console, 'error').mockImplementation(() => undefined);
  });

  it('should match the snapshot', () => {
    // Given
    const errorMock = {
      message: 'any error with message mock',
      stack: 'any error with stack mock',
    };

    // When
    const { container } = render(
      <AppBoundaryComponent error={errorMock} resetErrorBoundary={expect.any(Function)} />,
    );

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should call useDocumentTitle with the correct title', () => {
    // When
    render(
      <AppBoundaryComponent
        error={new Error('any error with message mock')}
        resetErrorBoundary={expect.any(Function)}
      />,
    );

    // Then
    expect(useDocumentTitle).toHaveBeenCalledOnce();
    expect(useDocumentTitle).toHaveBeenCalledWith('FranceConnect - Erreur');
  });
});
