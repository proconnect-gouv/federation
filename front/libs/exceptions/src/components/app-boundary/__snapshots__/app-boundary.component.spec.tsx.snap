// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBoundaryComponent should match the snapshot 1`] = `
<div>
  <div
    class="sticky-body"
  >
    <div
      class="sticky-content"
    >
      <header
        class="fr-header"
        role="banner"
      >
        <div
          class="fr-header__body"
        >
          <div
            class="fr-container"
          >
            <div
              class="fr-header__body-row"
            >
              <div
                class="fr-header__brand fr-enlarge-link"
              >
                <div
                  class="fr-header__brand-top"
                >
                  <div
                    class="fr-header__logo"
                  >
                    <a
                      aria-disabled="true"
                      href="/"
                      style="cursor: pointer;"
                      title="R√©publique Fran√ßaise"
                    >
                      <p
                        class="fr-logo"
                      >
                        R√©publique
                        <br />
                        Fran√ßaise
                      </p>
                    </a>
                  </div>
                  <div
                    class="fr-header__operator"
                  >
                    <img
                      alt="FranceConnect"
                      class="fr-responsive-img"
                      src="function SvgLogoFranceConnect(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgLogoFranceConnect.name})
  );
}"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <main
        class="fr-container fr-my-8w"
      >
        <h1>
          Oooops, Something went wrong ! ü§∑
        </h1>
        <div
          class="fr-alert fr-alert--error fr-alert--md"
        >
          <h3
            class="fr-alert__title"
          >
            any error with message mock
          </h3>
          <ul>
            <li>
              any error with stack mock
            </li>
          </ul>
        </div>
      </main>
    </div>
    <footer
      class="sticky-footer fr-footer"
      id="footer"
      role="contentinfo"
    >
      <div
        class="fr-container"
      >
        <div
          class="fr-footer__body"
        >
          <div
            class="fr-footer__brand fr-enlarge-link"
          >
            <p
              class="fr-logo"
            >
              R√©publique
              <br />
              Fran√ßaise
            </p>
            <a
              aria-disabled="true"
              class="fr-footer__brand-link"
              href="/"
              style="cursor: pointer;"
              title="R√©publique Fran√ßaise"
            >
              <img
                alt="FranceConnect"
                class="fr-footer__logo fr-responsive-img"
                src="function SvgLogoFranceConnect(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgLogoFranceConnect.name})
  );
}"
                style="height: 90px; max-height: 90px;"
              />
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
